.github/workflows/build.yml

```diff
@@ -22,28 +22,29 @@ jobs:
           - "faq"
           - "storage"
           - "cli"
+          - "spatial"
     timeout-minutes: 15
-
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: recursive
           fetch-depth: 0
 
-      - uses: actions/setup-python@v5
-        if: ${{ matrix.group == 'storage' }}
+      - uses: dorny/paths-filter@v3
```

Comment:
Interesting! I didn't know about this before.

Is it correct that you chose this over the native working-directory trigger because you wanted keep a single `build.yml`?

Reply:
But does this actually work? The last CI run ran all jobs. I'd have expected to only run the "spatial" job.

<img width="496" alt="image" src="https://github.com/user-attachments/assets/e399790a-9d2d-4f02-ae59-8f092a5810e4" />

Reply:
The idea is to only run the spatialdata tests when there is a change to spatialdata-related code. No other tests should then run.

Let's discuss tomorrow!

Reply:
> Is it correct that you chose this over the native working-directory trigger because you wanted keep a single build.yml?

For now! See https://github.com/laminlabs/lamindb/issues/2303

> But does this actually work? The last CI run ran all jobs. I'd have expected to only run the "spatial" job
> The idea is to only run the spatialdata tests when there is a change to spatialdata-related code. No other tests should then run.

The last CI is supposed to run all jobs with the current way I've designed it. It touched more than just the spatial code. I hear you that you want to exclusively only run parts that were touched and this will require more thinking - especially because some code pieces here are not isolated. For example, touching the other Curators can totally break the SpatialdataCurator. Hence, the spatial CI is triggered if any of the Curator code is touched.

Reply:
Ok, if this is how it behaves it's all great!

I'll play with this. I think it's very important that this is well organised.